src/jit/lsraarm64.cpp

```diff
@@ -994,10 +995,44 @@ void LinearScan::TreeNodeInfoInitSIMD(GenTreeSIMD* simdTree, TreeNodeInfo* info)
 void LinearScan::TreeNodeInfoInitHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, TreeNodeInfo* info)
 {
     NamedIntrinsic intrinsicID = intrinsicTree->gtHWIntrinsicId;
-    info->srcCount += GetOperandInfo(intrinsicTree->gtOp.gtOp1);
-    if (intrinsicTree->gtGetOp2IfPresent() != nullptr)
+
+    GenTreeArgList* argList = nullptr;
+    GenTree*        op1     = intrinsicTree->gtOp.gtOp1;
+    GenTree*        op2     = intrinsicTree->gtOp.gtOp2;
+
+    if (op1->OperIs(GT_LIST))
     {
-        info->srcCount += GetOperandInfo(intrinsicTree->gtOp.gtOp2);
+        argList = op1->AsArgList();
+        op1     = argList->Current();
+        op2     = argList->Rest()->Current();
+
+        for (GenTreeArgList* list = argList; list != nullptr; list = list->Rest())
+        {
+            info->srcCount += GetOperandInfo(list->Current());
+        }
+    }
+    else
+    {
+        info->srcCount += GetOperandInfo(op1);
```

Comment:
Is ARM not going to have 0 operand nodes or just not yet?

Reply:
I do not see a reason.  Unless we need to support barriers.

Reply:
> Unless we need to support barriers.
x86 added `StoreFence`, `LoadFence`, and `MemoryFence`. It also has a couple of helper methods (such as `Sse.SetZeroVector128`) which takes 0 args.

Reply:
Separate PR if needed.

Reply:
Sounds fine to me.